src/Crate/PDO/PDOStatement.php

```diff
@@ -202,7 +202,10 @@ public function execute($input_parameters = null)
             if (is_int($parameter) && $zero_based) {
                 $parameter++;
             }
-            $this->bindValue($parameter, $value);
+            // to correctly save arrays and objects, we need to pass
```

Comment:
Dne 12. 05. 2015 ob 13:24 je Marco Pivetta zapisal(a):
> In src/Crate/PDO/PDOStatement.php
> <https://github.com/crate/crate-pdo/pull/21#discussion_r30124401>:
>
>> @@ -202,7 +202,10 @@ public function execute($input_parameters = null)
>>              if (is_int($parameter) && $zero_based) {
>>                  $parameter++;
>>              }
>> -            $this->bindValue($parameter, $value);
>> +            // to correctly save arrays and objects, we need to pass
>
> Note that I'm just saying that |$stmt->execute($arrayOfValues);| should
> not be executed with typed parameters.
>
> If you have parameter types, then loop and use |bindValue|, because the
> PDO API simply doesn't expose something "safe to use" anyway.

Well, it really seems wrong to just fix all parameters as strings. It 
produces errors where one doesn't expect to find them.

You might add a documentation info about this, I guess we won't be the 
only ones sending parameters through execute() function.

You can NACK the patch now, sorry to bother.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crate/crate-pdo/pull/21/files#r30124401>.
>